lightning/src/ln/channelmanager.rs

```diff
@@ -3104,7 +3104,17 @@ impl<ChanSigner: ChannelKeys, M: Deref + Sync + Send, T: Deref + Sync + Send, K:
 						}
 					}
 				}
-				if channel.is_funding_initiated() && channel.channel_monitor().would_broadcast_at_height(height, &self.logger) {
+				#[cfg(any(test, feature = "fuzztarget"))]
+				// In testing/fuzzing, check that the channel's would_broadcast_at_height
+				// implementation is always identical to the ChannelMonitor equivalent.
+				let funding_initiated = channel.is_funding_initiated();
+				#[cfg(any(test, feature = "fuzztarget"))]
+				for i in height..height + 10 {
+					assert_eq!(funding_initiated && channel.monitor_would_broadcast_at_height(i, &self.logger),
+										 funding_initiated && self.monitor.get_monitor_would_broadcast(&channel.get_funding_txo().unwrap(), i));
```

Comment:
Ok, sg -- FYI, I lowered it to 10 because `144` was making the fuzz tests run ~80 minutes on GH Actions (up from ~15 minutes), and about an hour locally (up from ~5 minutes, on my computer).

Reply:
Hmm, would be good to figure out why its so slow - it may be that reducing allocations inside the Channel call would improve it.

Reply:
Unfortunately the fuzz tests still seem to be over an hour on GH actions. From my testing, the time increase seems correlated to the `for i in height..height + X`'s `X` variable here.

Do you think there'd be significant speedup if, say, the HTLCs were checked in one pass instead of two? Otherwise, maybe we could compromise with a number in the middle? Though FWIW maybe it's not such a big deal to have slow CI runs.

Reply:
Hmm, right, may just be amount of work - probably the most immediate thing to try is figure out which fuzzer it is (chanmon_consistency, probably) and reduce the number of block_connected calls (L326) that are made no matter the input.

Reply:
Ok cool, I halved the number of `block_connected` calls as you say, and that halved the fuzz runtime. Do you think ~40 minutes is OK for GH actions, or should we go even lower?

Reply:
Less about the runtime in GH, more about making sure fuzzing is quick enough to find issues - I think we can set the block_connected calls there to whatever is required to lock in a channel, instead of an arbitrary (and high) number like 50.

Reply:
Cool, the runtime is totally back to normal range now so the fuzzers should be about as fast as they were before.